docs/topics/auth/username.txt

```diff
@@ -0,0 +1,140 @@
+.. _username-validation:
+
+=============================
+Username Validation in Django
+=============================
+
+.. module:: django.contrib.auth.username_validation
+
+.. versionadded:: 3.2
+
+Django offers pluggable username validation. You can configure multiple
+username validators at the same time. A few validators are included in 
+Django, but you can write your own as well.
+
+Each username validator must provide a help text to explain the requirements to
+the user, validate a given username and return an error message if it does not
+meet the requirements. Validators can also have optional settings to fine tune
+their behavior.
+
+Validation is controlled by the :setting:`AUTH_USERNAME_VALIDATORS` setting.
+The default for the setting is an ``UnicodeUsernameValidator`` which
+validates whether given username contains English letters, numbers and @/./+/-/_
+characters or not.
```

Comment:
instead of "or not" it would be simpler to just say "whether the given username contains only English letters, numbers, and ``@``,``.``,``+``,``-``,``_`` characters.